### PR TITLE
chore: add dictionary fetch getbatch setbatch

### PR DESCRIPTION
**Pull request by poppoerika**
Added dictionary fetch, getbatch, setbatch APIs along with tests for those.

Ticket #17

### CODE REVIEW

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php [hunk 1]
```diff
@@ -837,3 +848,152 @@ class CacheDictionaryDeleteResponseError extends CacheDictionaryDeleteResponse
...
+        if ($this->isError()) {
+            return $this;
+        }
+        return null;
+    }
+}
+
+class CacheDictionaryFetchResponseHit extends CacheDictionaryFetchResponse
+{
+    private RepeatedField $items;
```
Comment (pgautier404) [inline src/Cache/CacheOperationTypes/CacheOperationTypes.php:881]: Is there any reason to hold onto these `items`?

Comment (poppoerika) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php:881]: Nope! Thank you for pointing it out.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php [hunk 2]
```diff
@@ -787,11 +793,16 @@ class CacheDictionaryGetResponseHit extends CacheDictionaryGetResponse
...
-        $this->value = $response->getFound()->getItems()[0]->getCacheBody();
-
+        if (!is_null($response) && is_null($cacheBody)) {
+            parent::__construct();
+            $this->value = $response->getFound()->getItems()[0]->getCacheBody();
+        }
+        if (is_null($response) && !is_null($cacheBody)) {
+            parent::__construct();
+            $this->value = $cacheBody;
+        }
```
[the remark this answers is not in the document]
Comment (poppoerika) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php:805]: 👍

File: src/Utilities/_DataValidation.php [hunk 3]
```diff
@@ -65,6 +65,29 @@ function validateValueName(string $valueName): void
     }
 }
 
+if (!function_exists('validateItems')) {
+    function validateItems(array $items): void
+    {
+        foreach ($items as $item) {
```
Comment (pgautier404) [inline src/Utilities/_DataValidation.php:71]: Can you also add a check to make sure the array itself isn't empty, please?

File: src/Utilities/_DataValidation.php [hunk 4]
```diff
@@ -65,6 +65,29 @@ function validateValueName(string $valueName): void
...
+            }
+        }
+
+    }
+}
+
+if (!function_exists('validateFieldsKeys')) {
+    function validateFieldsKeys(array $items): void
+    {
+        foreach ($items as $field => $value) {
```
Comment (pgautier404) [inline src/Utilities/_DataValidation.php:83]: Same as above-- please make sure `$items` isn't empty before trying to iterate over it.

File: tests/Cache/CacheClientTest.php [hunk 5]
```diff
@@ -40,10 +40,10 @@ public function setUp(): void
...
     }
 
-    public function tearDown(): void
-    {
-        $this->client->deleteCache($this->TEST_CACHE_NAME);
-    }
+//    public function tearDown(): void
+//    {
+//        $this->client->deleteCache($this->TEST_CACHE_NAME);
+//    }
```
Comment (pgautier404) [inline tests/Cache/CacheClientTest.php:46]: Please delete this comment block.

File: src/Utilities/_DataValidation.php (hunk 3 above)
Comment (poppoerika) [reply src/Utilities/_DataValidation.php:71]: I think it should be covered in this check?
If `$array = [""]`, this should throw `InvalidArgumentError`.
And that is checked in [this test](https://github.com/momentohq/client-sdk-php/pull/29/files#diff-0cf06050f3f70b72b13887aec2aaca681da458ce829b4335c999fcab3fd4c369R1118).
But please let me know if misunderstand your request!

Comment (poppoerika) [reply src/Utilities/_DataValidation.php:71]: Oh I got it! `$array = []` 👍 
Please ignore my comment above.

File: tests/Cache/CacheClientTest.php [hunk 6]
```diff
@@ -933,4 +933,224 @@ public function testDictionaryDeleteHappyPath()
...
+        $contentDictionary = [$field1 => $value1, $field2 => $value2];
+        $response = $this->client->dictionarySet($this->TEST_CACHE_NAME, $dictionaryName, $field1, $value1, true, 10);
+        $this->assertNotNull($response->asSuccess());
+        $response = $this->client->dictionarySet($this->TEST_CACHE_NAME, $dictionaryName, $field2, $value2, true, 10);
+        $this->assertNotNull($response->asSuccess());
+
+        $response = $this->client->dictionaryFetch($this->TEST_CACHE_NAME, $dictionaryName);
+        $this->assertNotNull($response->asHit());
+        $this->assertEquals($response->asHit()->dictionary(), $contentDictionary);
+        $this->assertSame($response->asHit()->dictionary(), $contentDictionary);
```
Comment (pgautier404) [inline tests/Cache/CacheClientTest.php:975]: I don't think we need this `assertSame`, but let me know if you had a particular reason for adding it.

File: tests/Cache/CacheClientTest.php (hunk 5 above)
Comment (poppoerika) [reply tests/Cache/CacheClientTest.php:46]: Oh, that's right!

File: tests/Cache/CacheClientTest.php (hunk 6 above)
Comment (poppoerika) [reply tests/Cache/CacheClientTest.php:975]: That's true. We're not asserting to check two objects are equal and reference the same object, so I will remove it.

**Review by pgautier404 — APPROVED** [review]
LGTM!

**Pull request merged**